src/coreComponents/physicsSolvers/SolverBase.cpp

```diff
@@ -458,162 +562,223 @@ real64 SolverBase::nonlinearImplicitStep( real64 const & time_n,
     if( dtAttempt > 0 )
     {
       resetStateToBeginningOfStep( domain );
+      resetConfigurationToBeginningOfStep( domain );
     }
 
-    // keep residual from previous iteration in case we need to do a line search
-    real64 lastResidual = 1e99;
-    integer & newtonIter = m_nonlinearSolverParameters.m_numNewtonIterations;
-    real64 scaleFactor = 1.0;
+    // it's the simplest configuration that can be attempted whenever Newton's fails as a last resource.
+    bool attemptedSimplestConfiguration = false;
 
-    // main Newton loop
-    for( newtonIter = 0; newtonIter < maxNewtonIter; ++newtonIter )
+    // Configuration loop
+    for( configurationLoopIter = 0; configurationLoopIter < maxConfigurationIter; ++configurationLoopIter )
     {
-      GEOSX_LOG_LEVEL_RANK_0( 1, GEOSX_FMT( "    Attempt: {:2}, NewtonIter: {:2}", dtAttempt, newtonIter ) );
-
-      // zero out matrix/rhs before assembly
-      m_localMatrix.zero();
-      m_rhs.zero();
-
+      if( getLogLevel() >= 1 )
```

Comment:
Maybe simply call `outputConfigurationStatistics( domain );` and let each solver decide what it wants to display depending on the appropriate `logLevel`?

Reply:
Yeah, this is probably a good idea.